test/integration/targets/ec2_group/tasks/main.yml

```diff
@@ -343,10 +368,45 @@
            - 'result.changed'
            - 'not result.group_id'
 
-  always:
+    - name: create a VPC
+      ec2_vpc_net:
+        name: "{{ resource_prefix }}-vpc"
+        state: present
+        cidr_block: "10.232.232.128/26"
+        ec2_region: '{{ec2_region}}'
+        ec2_access_key: '{{ec2_access_key}}'
+        ec2_secret_key: '{{ec2_secret_key}}'
+        security_token: '{{security_token}}'
+        tags:
+          Name: "{{ resource_prefix }}-vpc"
+          Description: "Created by ansible-test"
+      register: vpc_result
 
-    # ============================================================
-    - name: test state=absent (expected changed=false)
+    - name: create security group in the VPC
+      ec2_group:
+        name: '{{ec2_group_name}}'
+        description: '{{ec2_group_description}}'
+        ec2_region: '{{ec2_region}}'
+        ec2_access_key: '{{ec2_access_key}}'
+        ec2_secret_key: '{{ec2_secret_key}}'
+        security_token: '{{security_token}}'
+        vpc_id: '{{ vpc_result.vpc.id }}'
+        state: present
+        rules:
+        - proto: "tcp"
+          from_port: 8182
+          to_port: 8182
```

Comment:
It appears between 2.3 and now we lost the functionality provided by https://github.com/ansible/ansible/pull/20806. I suspect that also happened during the boto -> boto3 transition. Let me know if you want to do that or if you want me to make a follow up pr. Either is fine, but that would be another thing we need a test for/need to re-fix.